src/OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.cpp

```diff
@@ -0,0 +1,155 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+StateBuilder::StateBuilder(
+    const Shared<const Frame>& aFrameSPtr, const Array<Shared<const CoordinatesSubset>>& aCoordinatesSubsetsArray
+)
+    : frameSPtr_(aFrameSPtr),
+      coordinatesBrokerSPtr_(std::make_shared<CoordinatesBroker>(CoordinatesBroker(aCoordinatesSubsetsArray)))
+{
+}
+
+StateBuilder::StateBuilder(
+    const Shared<const Frame>& aFrameSPtr, const Shared<const CoordinatesBroker>& aCoordinatesBrokerSPtr
+)
+    : frameSPtr_(aFrameSPtr),
+      coordinatesBrokerSPtr_(aCoordinatesBrokerSPtr)
+{
+}
+
+bool StateBuilder::operator==(const StateBuilder& aStateBuilder) const
+{
+    if ((!this->isDefined()) || (!aStateBuilder.isDefined()))
+    {
+        return false;
+    }
+
+    if (this->frameSPtr_ != aStateBuilder.frameSPtr_)
+    {
+        return false;
+    }
+
+    for (const Shared<const CoordinatesSubset>& subset : this->coordinatesBrokerSPtr_->accessSubsets())
```

Comment:
It would not be enough to check that both brokers contain each other's subsets.
One would need to check that the subsets occupy the same index in each broker.

However, luckily for us, "somebody" went a bit overboard with the `CoordinatesBroker` `==` operator so that it hard checks all of this (dimension, subsets, and ordering), so all you need to do right after comparing the frame is:

```cpp
if (this -> coordinatesBrokerSPtr_ != aStateBuilder.coordinatesBrokerSPtr_)
{
    return false;
}
```